src/Components/Blazor/Build/src/Tasks/BlazorILLink.cs

```diff
@@ -0,0 +1,189 @@
+                if (!assemblyNames.Add(assemblyName))
+                {
+                    continue;
+                }
+
+                args.Append("-reference ")
+                    .AppendLine(Quote(assemblyPath));
+
+                var action = assembly.GetMetadata("action");
+                if ((action != null) && (action.Length > 0))
+                {
+                    args.Append("-p ");
+                    args.Append(action);
+                    args.Append(" ").AppendLine(Quote(assemblyName));
+                }
+            }
+
+            if (ReferenceAssemblyPaths != null)
+            {
+                foreach (var assembly in ReferenceAssemblyPaths)
+                {
+                    var assemblyPath = assembly.ItemSpec;
+                    var assemblyName = Path.GetFileNameWithoutExtension(assemblyPath);
+
+                    // Don't process references for which we already have
+                    // implementation assemblies.
+                    if (assemblyNames.Contains(assemblyName))
+                    {
+                        continue;
+                    }
+
+                    args.Append("-reference ").AppendLine(Quote(assemblyPath));
+
+                    // Treat reference assemblies as "skip". Ideally we
+                    // would not even look at the IL, but only use them to
+                    // resolve surface area.
+                    args.Append("-p skip ").AppendLine(Quote(assemblyName));
+                }
+            }
+
+            if (OutputDirectory != null)
+            {
+                args.Append("-out ").AppendLine(Quote(OutputDirectory.ItemSpec));
+            }
+
+            if (ClearInitLocals)
+            {
+                args.AppendLine("--enable-opt clearinitlocals");
+                if ((ClearInitLocalsAssemblies != null) && (ClearInitLocalsAssemblies.Length > 0))
+                {
+                    args.Append("-m ClearInitLocalsAssemblies ");
+                    args.AppendLine(ClearInitLocalsAssemblies);
+                }
+            }
+
+            if (ExtraArgs != null)
+            {
+                args.AppendLine(ExtraArgs);
+            }
+
+            if (DumpDependencies)
+            {
+                args.AppendLine("--dump-dependencies");
+            }
+
+            return args.ToString();
+        }
+
+        protected override bool HandleTaskExecutionErrors()
+        {
+            // Show a slightly better error than the standard ToolTask message that says "dotnet" failed.
+            Log.LogError($"ILLink failed with exited code {ExitCode}.");
+            return false;
+        }
+
+        protected override void LogEventsFromTextOutput(string singleLine, MessageImportance messageImportance)
+        {
+            if (!string.IsNullOrEmpty(singleLine) && singleLine.StartsWith("Unhandled exception.", StringComparison.Ordinal))
```

Comment:
This is a little iffy, but improves the error reporting experience during `dotnet build` and in the IDE:

![image](https://user-images.githubusercontent.com/174281/69469275-d45ca500-0d44-11ea-97ae-5d7bfad16b03.png)

![image](https://user-images.githubusercontent.com/174281/69469292-e76f7500-0d44-11ea-9b28-1a0a91ed2f09.png)